tools/analyze_issue.mjs

```diff
@@ -0,0 +1,272 @@
+        set(lines.slice(j, i).join('\n').trim());
+        j = i + 1;
+        set = value => {
+          puppeteerVersion = removeV(value);
+        };
+      } else if (lines[i].startsWith('### Node version')) {
+        set(lines.slice(j, i).join('\n').trim());
+        j = i + 1;
+        set = value => {
+          nodeVersion = removeV(value);
+        };
+      } else if (lines[i].startsWith('### Package manager version')) {
+        set(lines.slice(j, i).join('\n').trim());
+        j = i + 1;
+        set = value => {
+          packageManagerVersion = removeV(value);
+        };
+      } else if (lines[i].startsWith('### Package manager')) {
+        set(lines.slice(j, i).join('\n').trim());
+        j = i + 1;
+        set = value => {
+          packageManager = value.toLowerCase();
+        };
+      } else if (lines[i].startsWith('### Operating system')) {
+        set(lines.slice(j, i).join('\n').trim());
+        j = i + 1;
+        set = value => {
+          os = value.toLowerCase();
+        };
+      }
+    }
+    set(lines.slice(j, i).join('\n').trim());
+  }
+
+  let runsOn;
+  switch (os) {
+    case 'windows':
+      runsOn = 'windows-latest';
+      break;
+    case 'macos':
+      runsOn = 'macos-latest';
+      break;
+    case 'linux':
+      runsOn = 'ubuntu-latest';
+      break;
+    default:
+      console.log(`errorMessage=${ERROR_MESSAGES.unsupportedOs(os)}`);
+      throw new Error(`Unsupported OS: ${os}`);
+  }
+
+  if (!SUPPORTED_PACKAGE_MANAGERS.includes(packageManager)) {
+    console.log(
+      `errorMessage=${ERROR_MESSAGES.unsupportedPackageManager(packageManager)}`
+    );
+    throw new Error(`Unsupported package manager: ${packageManager}`);
+  }
+
+  if (!semver.valid(nodeVersion)) {
+    console.log(
+      `errorMessage=${ERROR_MESSAGES.invalidNodeVersion(nodeVersion)}`
+    );
+    throw new Error('Invalid Node version');
+  }
+  if (semver.lt(nodeVersion, LAST_SUPPORTED_NODE_VERSION)) {
+    console.log(
+      `errorMessage=${ERROR_MESSAGES.unsupportedNodeVersion(nodeVersion)}`
+    );
+    throw new Error(`Unsupported node version: ${nodeVersion}`);
+  }
+
+  if (!semver.valid(puppeteerVersion)) {
+    console.log(
+      `errorMessage=${ERROR_MESSAGES.invalidPuppeteerVersion(puppeteerVersion)}`
+    );
+    throw new Error(`Invalid puppeteer version: ${puppeteerVersion}`);
+  }
+  if (semver.lt(puppeteerVersion, LAST_SUPPORTED_PUPPETEER_VERSION)) {
+    console.log(
+      `errorMessage=${ERROR_MESSAGES.unsupportedPuppeteerVersion(
+        puppeteerVersion
+      )}`
+    );
+    throw new Error(`Unsupported puppeteer version: ${puppeteerVersion}`);
+  }
+
+  if (!semver.valid(packageManagerVersion)) {
+    console.log(
+      `errorMessage=${ERROR_MESSAGES.invalidPackageManagerVersion(
+        packageManagerVersion
+      )}`
+    );
+    throw new Error(
+      `Invalid package manager version: ${packageManagerVersion}`
+    );
+  }
+
+  console.log(`errorMessage=""`);
```

Comment:
I think we should just add `@actions/code` and use it here rather then using this to output method.

Reply:
For history -> Typo on my part `@actions/core` is package that simplifies working with GitHub Actions